src/test/clojure/xtdb/operator/scan_test.clj

```diff
@@ -820,59 +821,79 @@
-                                                         ["meets" [#inst "2020" #inst "2021"], [#inst "2021" #inst "2022"]]
-                                                         ["overlaps" [#inst "2020" #inst "2022"], [#inst "2021" #inst "2023"]]
-                                                         ["starts" [#inst "2020" #inst "2022"], [#inst "2020" #inst "2023"]]
-                                                         ["during" [#inst "2020" #inst "2023"], [#inst "2021" #inst "2022"]]
-                                                         ["finishes" [#inst "2020" #inst "2022"], [#inst "2021" #inst "2022"]]
-                                                         ["equals" [#inst "2020" #inst "2022"], [#inst "2020" #inst "2022"]]]
-
-                               :let [[range1 range2] (case dir
-                                                       :normal [range1 range2]
-                                                       :inverse [range2 range1])]]
-                           {:interval interval, :dir dir
-                            :range1 range1, :range2 range2}))
-
-          tx1 [(into [:put-docs :foo]
-                     (map-indexed (fn [idx {[vf vt] :range1}]
-                                    {:xt/id idx, :version 1
-                                     :xt/valid-from vf, :xt/valid-to vt}))
-                     intervals)]
-
-          tx2 [(into [:put-docs :foo]
-                     (map-indexed (fn [idx {[vf vt] :range2}]
-                                    {:xt/id idx, :version 2,
-                                     :xt/valid-from vf, :xt/valid-to vt}))
-                     intervals)]]
-
-      (xt/execute-tx n1 tx1)
-      (xt/execute-tx n2 tx1)
-      (tu/finish-block! n2)
-      (c/compact-all! n2)
-      (xt/execute-tx n1 tx2)
-      (xt/execute-tx n2 tx2)
-      (tu/finish-block! n2)
-      (c/compact-all! n2)
-
-      (doseq [[idx {:keys [interval dir range1 range2]}] (map vector (range) intervals)]
-        (t/testing (format "%s interval: '%s'" (name dir) interval)
-          (doseq [date (into #{(Date. Long/MIN_VALUE) (Date. Long/MAX_VALUE)}
-                             (comp cat
-                                   (mapcat (fn [^Date date]
-                                             (for [^long delta (range -2 3)]
-                                               (Date. (+ (.getTime date) delta))))))
-                             [range1 range2])]
-            (t/testing (format "date: '%s'" (pr-str date))
-              (doseq [q ["SELECT * FROM foo FOR VALID_TIME AS OF ? WHERE _id = ?"
-                         "SELECT *, _valid_from, _valid_to FROM foo FOR VALID_TIME AS OF ? WHERE _id = ?"]]
-                (t/testing (format "query: '%s'" q)
-                  ;; params the wrong way around - see #4305
-                  (t/is (= (xt/q n1 [q idx date])
-                           (xt/q n2 [q idx date]))))))))))))
+  (doseq [bucketing  [RecencyPartition/WEEK RecencyPartition/YEAR]
```

Comment:
This test now runs in ~20 secs. Could take some factors if you find it too long.

Reply:
Could also make it an integration test.